citation/export_data.py

```diff
@@ -20,44 +23,52 @@ def create_output(items, values):
-    header = ["Count", "Id", "Publication Title", "Abstract", "Short Title", "Zotero Key", "Code Url",
+# Creating header for csv file
+def generate_csv_header():
+    header = ["Id", "Publication Title", "Abstract", "Short Title", "Zotero Key", "Code Url",
               "Date Published Text", "Date Accessed", "Archive", "Archive location", "Library catalog",
               "Call number", "Rights", "Extra", "Published Language", "Date Added", "Date Modified",
               "Zotero Date Added", "Zotero Date Modified", "Status", "Code Archive url", "Contact Email",
               "Author Comments", "Email Sent out", "Added by", "Assign Curator", "Conatct Author Name",
               "Resource type", "Is primary", "Journal Id", "doi", "Series text", "Series title", "Series", "Issue",
               "Volume", "ISSN", "pages", "Year of Publication", "Journal", "Notes", "Platform List"]
-    all_platforms = Platform.objects.all().values_list("name").order_by("name")
-    all_sponsors = Sponsor.objects.all().values_list("name").order_by("name")
-    header.extend(get_attribute_values(all_platforms))
+
+    header.extend(get_attribute_values(ALL_PLATFORMS))
     header.append("Sponsors List")
-    header.extend(get_attribute_values(all_sponsors))
-    publications = Publication.objects.filter(is_primary=True).prefetch_related('sponsors', 'platforms')
-    count = 1
-    writer.writerow(header)
+    header.extend(get_attribute_values(ALL_SPONSORS))
+    return header
+
+
+# Creating row for Csv file
+def generate_csv_row(pub):
+    platforms = get_attribute_values(list(pub.platforms.all().values_list("name")))
+    sponsors = get_attribute_values(list(pub.sponsors.all().values_list("name")))
+    notes = get_attribute_values(list(pub.note_set.all().values_list("text")))
+
+    platforms_output = generate_list(ALL_PLATFORMS, platforms)
+    sponsors_output = generate_list(ALL_SPONSORS, sponsors)
+
+    row = [pub.pk, pub.title, str(pub.abstract), pub.short_title, pub.zotero_key, pub.url,
+           pub.date_published_text, pub.date_accessed, pub.archive, pub.archive_location,
+           pub.library_catalog, pub.call_number, pub.rights, pub.extra, pub.published_language, pub.date_added,
+           pub.date_modified, pub.zotero_date_added, pub.zotero_date_modified, pub.status,
+           pub.code_archive_url, pub.contact_email, pub.author_comments,
+           pub.email_sent_count, pub.added_by, pub.assigned_curator, pub.contact_author_name,
+           pub.container.type, pub.is_primary, pub.container.id, pub.doi, pub.series_text,
+           pub.series_title, pub.series, pub.issue, pub.volume, pub.issn, pub.pages,
+           pub.container.date_added, pub.container.name, notes, platforms]
+    row.extend(platforms_output)
+    row.append(sponsors)
+    row.extend(sponsors_output)
+
+    return row
+
+
+def create_csv(writer):
+    publications = Publication.objects.filter(is_primary=True).prefetch_related('sponsors', 'platforms', 'note_set')
```

Comment:
This is fine for now but we should create a `PublicationQuerySet` that encapsulates this type of ORM logic into something like `Publication.objects.primary(*args, **kwargs)`